lib/stdlib/src/digraph.erl

```diff
@@ -230,7 +230,7 @@ in_neighbours(G, V) ->
       Edges :: [edge()].
 
 in_edges(G, V) ->
-    ets:select(G#digraph.ntab, [{{{in, V}, '$1'}, [], ['$1']}]).
+    [E || {{in, _}, E} <- ets:lookup(G#digraph.ntab, {in, V})].
```

Comment:
I may be missing something, but I believe you can still keep it as a `ets:select/2`, you'd need to wrap '$1' in `{const, '$1'}`.

Reply:
@josevalim I can't see why wrapping `'$1'` would help. Trying to wrap `{in, V}` doesn't work either because `const` tuples are not allowed in patterns. The following works:

    ets:select(G#digraph.ntab, [{'$1',
                                 [{'=:=', {element, 1, '$1'}, {const, {in, V}}}],
                                 [{element, 2, '$1'}]}]).

But that is not an improvement over using `ets:lookup/2` and a list comprehension because the guard expression will have to be evaluated for every record in the table.

Reply:
I might be missing something too, but wrapping the edge term (`$1`) won't help us, the `_` in `V` will still be interpreted as a wildcard. What we would like to do is to make ets interpret the `_` in the match-head as a literal. Reading  [Match Specifications in Erlang](https://erlang.org/doc/apps/erts/match_spec.html), it looks like we cannot use `const` in match heads, only in the guard, so:

```Erlang
    ets:select(G#digraph.ntab,
               [{{{in, V}, '$1'},
                 [{'=:=', {const, V}, {element, 2, {element, 1, '$_'}}}],
                 ['$1']}]).
``` 

would give us what we want. Considering that a digraph might have a large number of edges, this would probably have much worse overhead than the extra garbage we produce by the list comprehension digging out the edges. The `select` version will have to traverse the complete table and match against each edge. The  `lookup`-based implementation will only process the edges incident to `V`, and with `ntab` being a bag, I assume this won't require a complete scan.